File: nodejs/checkio/src/electronic_station/verify_anagrams.ts
import assert from "assert";

function verifyAnagrams(line1: string, line2: string): boolean {
	const sorted1 = line1
		.toLowerCase()
		.split("")
		.filter((char) => char !== " ")
		.sort()
		.join("");
	const sorted2 = line2
		.toLowerCase()
		.split("")
		.filter((char) => char !== " ")
		.sort()
		.join("");
	return sorted1 === sorted2;
}

console.log("Example:");
console.log(verifyAnagrams("Programming", "Gram Ring Mop"));

// These "asserts" are used for self-checking
assert.strictEqual(verifyAnagrams("Programming", "Gram Ring Mop"), true);
assert.strictEqual(verifyAnagrams("Hello", "Ole Oh"), false);
assert.strictEqual(verifyAnagrams("Kyoto", "Tokyo"), true);

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/electronic-station/verify-anagrams.py
def verify_anagrams(a: str, b: str) -> bool:
    # TODO:
    return sorted(a.lower().replace(" ", "")) == sorted(b.lower().replace(" ", ""))


print("Example:")
print(verify_anagrams("Programming", "Gram Ring Mop"))

assert verify_anagrams("Programming", "Gram Ring Mop") == True
assert verify_anagrams("Hello", "Ole Oh") == False
assert verify_anagrams("Kyoto", "Tokyo") == True

print("The mission is done! Click 'Check Solution' to earn rewards!")
